k8s/client.go

```diff
@@ -11,6 +11,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
```

Comment:
So, I read the source for that package, and yep not convinced we need this.  Its convenience comes at the cost of 1) a package dependency, b) reflection c) complexity.  For now I would either write a simple helper function that does the contains() search or use the standard lib functions.

Reply:
For instance, you could add a simple helper function here https://github.com/vmware-tanzu/crash-diagnostics/blob/a3c284543bb49d9158f2addcf4f0c2dccd85ab08/k8s/client.go#L260 to do the search.  Yes, it will be an N x M search, but probably faster, but you don't carry that extra baggage.

```go
func sliceContains(slice []string, values ...string) bool {
    for _, s := range slice{
        for _, val :=range values {
            // search
        }
    }
}
```

Reply:
I don't have strong opinions for using the `sets` package. I will re-write it to do the NxM search.